src/v/kafka/server/quota_manager.cc

```diff
@@ -344,17 +344,13 @@ ss::future<> quota_manager::do_gc(clock::time_point expire_threshold) {
         return res;
     };
 
-    auto reducer = [](const key_set& acc, const key_set& next) -> key_set {
-        key_set result;
-        // Note: std::set_intersection assumes the inputs are sorted and
-        // guarantees that the output is sorted
-        std::set_intersection(
-          acc.begin(),
-          acc.end(),
-          next.begin(),
-          next.end(),
-          std::back_inserter(result));
-        return result;
+    auto reducer = [](key_set acc, const key_set& next) -> key_set {
+        // In-place set intersection assumes that the inputs are sorted and
+        // guarantees that the output is also sorted
+        auto it = std::set_intersection(
+          acc.begin(), acc.end(), next.begin(), next.end(), acc.begin());
```

Comment:
[27.8.7.4.2](https://open-std.org/jtc1/sc22/wg21/docs/papers/2024/n4981.pdf) "Preconditions: The ranges [first1, last1) and [first2, last2) are sorted with respect to comp and
proj1 or proj2, respectively. The resulting range does not overlap with either of the original ranges."

...but I believe the implementation is actually safe. Maybe copy the implementation from libc++?